ci/docker/tizen/tools/cache-restore-mtime

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+# Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# After `gclient sync`, the mtime of all synchronized files will be changed
+# to newer than the cached out files. Therefore, this script changes the mtime
+# of all files to OLD time and changes only the mtime of the changed files
+# to NEW time.
```

Comment:
Why does this matter?

Actually the overall workflow is not very intuitive for me. For what do you restore mtimes and save checksums of files? Are both `gclient-reduce-deps` and `gclient-fast-sync` required and should they be separate files? (As far as I understand the both exist for the same reason: reducing the gclient sync time.) Why do you run `gclient sync -v --no-history --shallow` after `gclient-fast-sync`? (Isn't `gclient-fast-sync` supposed to replace `gclient sync`, assuming from its name?) Why is `gclient-fast-sync.py` alone a python script while other tools are written in shell script?

Reply:
1. The reason for `restoring mtimes` is for the incremental build. After `gclient sync`, the mtime of all synchronized files will be changed to newer than the cached out files. So ninja build will try to clean build always. to solve this problem, this script changes the mtime  of all files to OLD time except new changed files.
2. Yes, those scripts can be one. But I was difficult to set the name of that script....
3. Please see the test workflow file: https://github.com/WonyoungChoi/engine/blob/citest/.github/workflows/build.yml. 
```yaml
    - name: gclient sync
      run: |
        gclient-reduce-deps
        gclient-fast-sync
        gclient sync -v --no-history --shallow
```
4. `gclient-fast-sync.py` script uses some python classes of the `depot-tools` to parse DEPS file.

Reply:
how about `gclient-prepare-sync` for the merged one. looks like.. 
```yaml
    - name: gclient sync
      run: |
        gclient-prepare-sync --reduce-deps --shallow-sync
        gclient sync -v --no-history --shallow
```